lua/entities/gmod_tardis/modules/sh_tracking.lua

```diff
@@ -0,0 +1,497 @@
+    end
+
+    function ENT:ToggleTrackRotation()
+        self:SetTrackRotation(not self:GetTrackRotation())
+    end
+
+    ENT:AddHook("Initialize", "tracking", function(self)
+        self:SetTrackRotation(true)
+    end)
+
+    ENT:AddHook("HandleE2", "tracking", function(self, name, e2, ...)
+        local args = {...}
+        if name == "Track" and (args[1] == e2.player or TARDIS:CheckPP(e2.player, self)) then
+            local ent = args[1]
+            local success = self:SetTracking(ent, e2.player)
+            return success and 1 or 0
+        end
+    end)
+
+    ENT:AddHook("HandleE2", "tracking_get", function(self, name, e2)
+        if name == "GetTracking" then
+            return self:GetTracking() or NULL
+        end
+    end)
+
+    ENT:AddHook("FlightToggled", "tracking", function(self, on)
+        if not on then
+            self:SetTracking(nil, self:GetData("pilot"))
+        end
+    end)
+
+    local VECTOR_0_0_1 = Vector(0,0,1)
+
+    ENT:AddHook("PhysicsUpdate", "tracking", function(self, ph)
+        local ent = self:GetTracking()
+        if ent and not IsValid(ent) then
+            self:SetTracking(nil, self:GetData("pilot"))
+            return
+        elseif not ent then
+            return
+        end
+
+        local pos = self:GetPos()
+        local entPos = ent:GetPos()
+        local offset = self:GetData("tracking-offset-pos", Vector(0,0,0))
+        local yawoffset = self:GetData("tracking-offset-yaw", 0)
+        local pilot = self:GetData("pilot")
+        local phm = FrameTime()*66
+        local offsetforce = 5
+        local offsetrforce = 2
+        local spin = self:GetSpin()
+
+        if IsValid(pilot) then
+            local eye = pilot:GetTardisData("viewang")
+            if not eye then
+                eye = angle_zero
+            end
+            local fwd = eye:Forward()
+            local ri = eye:Right()
+
+            local fbinds = {
+                forward = TARDIS:IsBindDown(pilot,"flight-forward"),
+                backward = TARDIS:IsBindDown(pilot,"flight-backward"),
+                left = TARDIS:IsBindDown(pilot,"flight-left"),
+                right = TARDIS:IsBindDown(pilot,"flight-right"),
+                rotate = TARDIS:IsBindDown(pilot,"flight-rotate"),
+                up = TARDIS:IsBindDown(pilot,"flight-up"),
+                down = TARDIS:IsBindDown(pilot,"flight-down"),
+                boost = TARDIS:IsBindDown(pilot,"flight-boost"),
+            }
+
+            if fbinds.boost then
+                offsetforce = offsetforce*TARDIS:GetSetting("boost-speed")
+                offsetrforce = offsetrforce*TARDIS:GetSetting("boost-speed")
+            end
+
+            local adjustedOffset = Vector()
+            if fbinds.forward then
+                adjustedOffset:Add(fwd * offsetforce)
+            end
+            if fbinds.backward then
+                adjustedOffset:Add(fwd * -offsetforce)
+            end
```

Comment:
**BUG**: when tracking-rotation is off, forward-back and right-left controls sometimes swap or work at wrong angles. 
E.g. you press forward and go right

It happened to me while I was tracking a patrolling NPC switching directions

Reply:
UPD: it does indeed depend on the tracked entity's yaw

Reply:
https://github.com/MattJeanes/TARDIS/pull/969/commits/95f488928ae50fc1df4c7b08f5bd8484657c2a63

Reply:
Seems to be okay now, thanks!